#! /usr/bin/env python

PACKAGE='lanekeeping'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name             Type             Level                   Description                                   Default      Min        Max 
#  add the parameters that you want to change here 

gen.add("kp",            double_t,             0,                 "KP of the PID postion error"                  , 1,      0.1,    25) # 2
gen.add("ki",            double_t,             0,                 "KI of the PID postion error"                   , 1,      0.1,   25) # 1
gen.add("kd",            double_t,             0,                 "KD of the PID postion error"                   , 1.8,     0.1,    25) # 2
gen.add("bez_kp",            double_t,          0,             "KP of the PID postion error of bez curve"       , 0.5,      1,    25)
gen.add("bez_ki",            double_t,          0,             "KI of the PID postion error of bez curve"       , 1,      1,   25)
gen.add("bez_kd",            double_t,          0,             "KD of the PID postion error of bez curve"       , 1,      1,    25)
gen.add("speed",             int_t,             0,                 "Speed of the car"                              , 2,     1,    25)
gen.add("up_sat_value",     double_t,          0.0,                 "control_cmd up saturation value"             , 1.0,    0.1,    5.0)
gen.add("look_Ahead_dist",  int_t,           0,                      "look ahead distance "                        , 2,      0,    100) #2
gen.add("turn_left",        bool_t,         0,                      "turn left"                                    ,False)
gen.add("turn_right",       bool_t,         0,                      "turn right"                                  , False)
gen.add("x_start",          double_t,          0,                       "start x"                                    , 3,      0,    100)
gen.add("y_start",          double_t,          0,                       "start y"                                    , 0,      0,    100)




exit(gen.generate(PACKAGE, PACKAGE, "Lane_Keeping"))
# we should be good now !! :-)
